numpy/core/arrayprint.py

```diff
@@ -706,6 +706,26 @@ def _extendLine(s, line, word, line_width, next_line_prefix, legacy):
     return s, line
 
 
+def _extendLine_pretty(s, line, word, line_width, next_line_prefix, legacy):
+    words = word.splitlines()
+    line_length = len(line)
+    s, line = _extendLine(
+        s, line, words[0], line_width, next_line_prefix, legacy)
+    for word in words[1::]:
+        s += line.rstrip() + '\n'
+        needs_wrap = line_length + len(word) > line_width
+        if legacy != '1.13':
+            # don't wrap lines if it won't help
+            if line_length <= len(next_line_prefix):
+                needs_wrap = False
```

Comment:
I suspect if `legacy` is passe we shouldn't do this pretty wrapping at all.

Reply:
done